pkg/image/admission/imagepolicy/api/v1/types.go

```diff
@@ -0,0 +1,138 @@
+type ImagePolicyConfig struct {
+	unversioned.TypeMeta `json:",inline"`
+
+	// ExecutionRules determine whether the use of an image is allowed in an object with a pod spec.
+	// By default, these rules only apply to pods, but may be extended to other resource types.
+	// If all execution rules are negations, the default behavior is allow all. If any execution rule
+	// is an allow, the default behavior is to reject all.
+	ExecutionRules []ImageExecutionPolicyRule `json:"executionRules"`
+
+	// TODO: enable additional rule types
+	// ConsumptionRules are applied when creating resources with pod specs.
+	// ConsumptionRules []ImageConsumptionPolicyRule `json:"consumptionRules"`
+	// PlacementRules are applied when creating resources with pod specs.
+	// PlacementRules []ImagePlacementPolicyRule `json:"placementRules"`
+}
+
+// ImageExecutionPolicyRule determines whether a provided image may be used on the platform.
+type ImageExecutionPolicyRule struct {
+	ImageCondition `json:",inline"`
+
+	// Resolve indicates that images referenced by this resource must be resolved
+	Resolve bool `json:"resolve"`
+}
+
+// ImageCondition defines the conditions for matching a particular image source. The conditions below
+// are all required (logical AND). If Reject is specified, the condition is false if all conditions match,
+// and true otherwise.
+type ImageCondition struct {
+	// Name is the name of this policy rule for reference. It must be unique across all rules.
+	Name string `json:"name"`
+	// IgnoreNamespaceOverride prevents this condition from being overriden when the
+	// `alpha.image.policy.openshift.io/ignore-rules` is set on a namespace and contains this rule name.
+	IgnoreNamespaceOverride bool `json:"ignoreNamespaceOverride"`
+
+	// Reject indicates this rule is inverted - if all conditions below match, then this rule is considered to
+	// not match.
+	Reject bool `json:"reject"`
+
+	// OnResources determines which resources this applies to. Defaults to 'pods' for ImageExecutionPolicyRules.
+	OnResources []unversioned.GroupResource `json:"onResources"`
+
+	// MatchIntegratedRegistry will only match image sources that originate from the configured integrated
+	// registry.
+	MatchIntegratedRegistry bool `json:"matchIntegratedRegistry"`
+	// MatchRegistries will match image references that point to the provided registries
+	MatchRegistries []string `json:"matchRegistries"`
+
+	// AllowResolutionFailure allows the subsequent conditions to be bypassed if the integrated registry does
+	// not have access to image metadata (no image exists matching the image digest).
+	AllowResolutionFailure bool `json:"allowResolutionFailure"`
+
+	// MatchDockerImageLabels checks against the resolved image for the presence of a Docker label
+	MatchDockerImageLabels []ValueCondition `json:"matchDockerImageLabels"`
+	// MatchImageLabels checks against the resolved image for a label.
+	MatchImageLabels []ValueCondition `json:"matchImageLabels"`
```

Comment:
Forgot about that.

On Fri, Aug 5, 2016 at 8:36 AM, David Eads <notifications@github.com> wrote:

> In pkg/image/admission/imagepolicy/api/v1/types.go
> <https://github.com/openshift/origin/pull/8995#discussion_r73685115>:
>
> > +	OnResources []unversioned.GroupResource `json:"onResources"`
> > +
> > +	// MatchIntegratedRegistry will only match image sources that originate from the configured integrated
> > +	// registry.
> > +	MatchIntegratedRegistry bool `json:"matchIntegratedRegistry"`
> > +	// MatchRegistries will match image references that point to the provided registries
> > +	MatchRegistries []string `json:"matchRegistries"`
> > +
> > +	// AllowResolutionFailure allows the subsequent conditions to be bypassed if the integrated registry does
> > +	// not have access to image metadata (no image exists matching the image digest).
> > +	AllowResolutionFailure bool `json:"allowResolutionFailure"`
> > +
> > +	// MatchDockerImageLabels checks against the resolved image for the presence of a Docker label
> > +	MatchDockerImageLabels []ValueCondition `json:"matchDockerImageLabels"`
> > +	// MatchImageLabels checks against the resolved image for a label.
> > +	MatchImageLabels []ValueCondition `json:"matchImageLabels"`
>
> I still think that if we're going to have label selectors, we should use
> the unversioned.LabelSelector.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8995/files/f96f5e56ba70c7c0f24f875621efdf1ef45ce3c0#r73685115>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p8ytpQ3r0Pr2ZqeQnKorjEfhdtOkks5qcy5WgaJpZM4IlDlb>
> .
>